app/controllers/journeys/expenses/goodsToSellOrUse/DisallowableGoodsToSellOrUseAmountController.scala

```diff
@@ -17,67 +17,59 @@
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
 import utils.MoneyUtils.formatMoney
 import views.html.journeys.expenses.goodsToSellOrUse.DisallowableGoodsToSellOrUseAmountView
 
 import javax.inject.{Inject, Singleton}
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.ExecutionContext
 
 @Singleton
 class DisallowableGoodsToSellOrUseAmountController @Inject() (override val messagesApi: MessagesApi,
-                                                              selfEmploymentService: SelfEmploymentService,
-                                                              navigator: ExpensesNavigator,
+                                                              val controllerComponents: MessagesControllerComponents,
+                                                              service: SelfEmploymentService,
                                                               identify: IdentifierAction,
                                                               getData: DataRetrievalAction,
                                                               requireData: DataRequiredAction,
                                                               formProvider: DisallowableGoodsToSellOrUseAmountFormProvider,
-                                                              val controllerComponents: MessagesControllerComponents,
                                                               view: DisallowableGoodsToSellOrUseAmountView)(implicit ec: ExecutionContext)
     extends FrontendBaseController
     with I18nSupport {
 
-  def onPageLoad(taxYear: TaxYear, businessId: BusinessId, mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) async {
-    implicit request =>
-      request.userAnswers.get(GoodsToSellOrUseAmountPage, Some(businessId)) match {
-        case None => Future.successful(Redirect(routes.JourneyRecoveryController.onPageLoad()))
-        case Some(goodsAmount) =>
-          val preparedForm =
-            request.userAnswers.get(DisallowableGoodsToSellOrUseAmountPage, Some(businessId)) match {
-              case None        => formProvider(request.userType, goodsAmount)
-              case Some(value) => formProvider(request.userType, goodsAmount).fill(value)
-            }
+  private val page = DisallowableGoodsToSellOrUseAmountPage
 
-          Future.successful(Ok(view(preparedForm, mode, request.userType, taxYear, businessId, formatMoney(goodsAmount))))
-      }
+  def onPageLoad(taxYear: TaxYear, businessId: BusinessId, mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
+    implicit request =>
+      (for {
+        allowableAmount <- request.valueOrRedirectDefault(GoodsToSellOrUseAmountPage, businessId)
+        form = fillForm(page, businessId, formProvider(request.userType, allowableAmount))
+      } yield Ok(view(form, mode, request.userType, taxYear, businessId, formatMoney(allowableAmount)))).merge
```

Comment:
Not wrong by any means, but worth considering just explicitly using map if we are not sequencing any monadic operations.

Reply:
Yep, think you're right. Will change them